swift/extractor/SwiftDispatcher.h

```diff
@@ -83,6 +83,11 @@ class SwiftDispatcher {
     return label;
   }
 
+  template <typename E, typename = std::enable_if_t<!std::is_pointer_v<E>>>
```

Comment:
Nit: We can still call this by specifying the SFINAE type argument:
```
int* foo = nullptr;
assignNewLabel<int*, void>(foo)
```

See https://gcc.godbolt.org/z/borzbbxfG for more examples.

One way to get around that is to use the `enable_if_t` in a non-type template parameter, so the caller only has control of the value, not the type (which then fails to substitute regardless of the user-specified value):

```
template <typename E, std::enable_if_t<!std::is_pointer_v<E>>* = nullptr>
```

Reply:
I don't like that as much because it makes the return type less obvious. The alternative of putting it in the template parameter list makes it less obvious what the "real" template parameters are, but I think that's less important? Happy to concede that it's just a matter of taste.

Really what we need is concepts, but given that even our C++17 support is incomplete, that might be a stretch...

Reply:
yeah, I wish we had concepts as well, we could just write `template<not_a_pointer E>` with the appropriate concept definition, and it would be readable and provide meaningful error messages when needed.

I'll go with the non-type parameter trick :+1: